lib/vector-core/src/config/proxy.rs

```diff
@@ -105,32 +107,36 @@ impl ProxyConfig {
         }
     }
 
-    fn http_intercept(&self) -> Intercept {
-        self.interceptor().intercept("http")
-    }
-
-    fn http_proxy(&self) -> Result<Option<Proxy>, InvalidUri> {
-        self.http
+    fn build_proxy(
+        &self,
+        proxy_scheme: &'static str,
+        proxy_url: &Option<String>,
+    ) -> Result<Option<Proxy>, InvalidUri> {
+        proxy_url
             .as_ref()
             .map(|url| {
-                url.parse()
-                    .map(|parsed| Proxy::new(self.http_intercept(), parsed))
+                url.parse().map(|parsed| {
+                    let mut proxy = Proxy::new(self.interceptor().intercept(proxy_scheme), parsed);
+                    if let Ok(authority) = Url::parse(url) {
+                        let username = authority.username();
+                        if !username.is_empty() {
```

Comment:
I think we could just check for the existence of `password`? I think I've seen cases where a username isn't used, but just a password. That is, just remove this `is_empty` check and leave the ` if let Some(password) = authority.password()` below.

Reply:
I was wondering about the case. It wasn't clear what should happen if the username is missing, but I think this makes sense.

Reply:
done!